src/ImageSharp/Formats/Jpeg/JpegDecoderCore.cs

```diff
@@ -1055,18 +1127,18 @@ private void ProcessStartOfFrameMarker(BufferedReadStream stream, int remaining,
             // 2 byte: Width
             int frameWidth = (this.temp[3] << 8) | this.temp[4];
 
-            // Validate: width/height > 0 (they are upper-bounded by 2 byte max value so no need to check that)
+            // Validate: width/height > 0 (they are upper-bounded by 2 byte max value so no need to check that).
             if (frameHeight == 0 || frameWidth == 0)
```

Comment:
```suggestion
            if ((frameHeight | frameWidth) == 0)
```
produces [less code](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIAYACY8gOgCUBXAOwwEt8YLAMIR8AB14AbGFADKMgG68wMXAG4aNYgGYmpBgDEIEAELYoNAN40GtpruDHJh4+QAUvHgwBmUbAIAJGF4AcwALDDQGTwwfPwEAdV4AEwwwgEoGAF4APjj/GCDQiOyshkZCQnzElLTS8o1qO3sGRwhnIwhSDy9fAqLwyOje+Jgk1IzsvLc+wODBhirZsdrJrLK6RoBfIA) and avoids a branch.

Note: current JIT will do this optimizaiton for use, but older runtimes miss that optimization.